src/ty/ty_check.rs

```diff
@@ -156,7 +161,7 @@ impl<'a> Context<'a> {
                 }
             }
             Term::PrimFn(prim) => match prim {
-                _ => todo!("There aren't any primitives yet!"),
+                Primitive::Print => Ty::Unit,
```

Comment:
Maybe we can fix this in the same way as @seanchen1991 did before we had binary operations as `Term`s inside the MIR.

You can add an special case to the rule for `Term::App(t1, t2)` when `t1` is `Term::PrimFn(Primitive::Print)` and type it as `Ty::Unit`. 

And then you can change this arm to `unreachable!()`

Reply:
Moved type check for primitives to own method